### PR TITLE
LIA: Use most-infeasible branching heuristic

### PR DESCRIPTION
**Pull request by blishko**
When Simplex model needs fixing because of integer domain, let's be a bit more clever than just branching on the first variable we find.

There are a couple of heuristics that could be used, most-infeasible branching seems like a good one.
It considers all integer variables that needs fixing and chooses the one with largest distance from an integer point (i.e., minimum from the two distances to the two closest points: floor and ceiling).

### CODE REVIEW

**Review by aehyvari — COMMENTED** [review]
It looks good!  I tried to refactor the heuristic to a different function.  Which one is better?

File: src/tsolvers/lasolver/LASolver.cc [hunk 1]
```diff
@@ -348,69 +344,52 @@ void LASolver::declareAtom(PTRef leq_tr)
     setKnown(leq_tr);
 }
 
+LVRef LASolver::splitOnMostInfeasible(vec<LVRef> const & varsToFix) const {
+    opensmt::Real maxDistance = 0;
+    LVRef chosen;
```
Comment (blishko) [inline src/tsolvers/lasolver/LASolver.cc:349]:
```suggestion
    LVRef chosen = LVRef_Undef;
```

**Review by blishko — COMMENTED** [review]
I think we keep it in separate function, as you suggested.

File: src/tsolvers/lasolver/LASolver.cc (hunk 1 above)
Comment (aehyvari) [reply src/tsolvers/lasolver/LASolver.cc]: Thanks, I noticed!

**Review by aehyvari — APPROVED** [review]

**Pull request merged**